hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/pauses/HapiSpecWaitUntilNextBlock.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.spec.utilops.pauses;
+
+import static com.hedera.services.bdd.junit.hedera.ExternalPath.BLOCK_STREAMS_DIR;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoTransfer;
+import static com.hedera.services.bdd.spec.transactions.crypto.HapiCryptoTransfer.tinyBarsFromTo;
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.suites.HapiSuite.FUNDING;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+
+import com.hedera.node.app.blocks.impl.FileBlockItemWriter;
+import com.hedera.services.bdd.junit.support.BlockStreamAccess;
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.spec.utilops.UtilOp;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class HapiSpecWaitUntilNextBlock extends UtilOp {
+    private static final Logger log = LogManager.getLogger(HapiSpecWaitUntilNextBlock.class);
+    private static final String BLOCK_FILE_EXTENSION = ".blk";
+    private static final String COMPRESSED_BLOCK_FILE_EXTENSION = BLOCK_FILE_EXTENSION + ".gz";
+    private static final String MARKER_FILE_EXTENSION = ".mf";
+    private static final Duration POLL_INTERVAL = Duration.ofMillis(100);
+    private static final Duration BACKGROUND_TRAFFIC_INTERVAL = Duration.ofMillis(1000);
+    private static final Duration TIMEOUT = Duration.ofSeconds(10);
+
+    private boolean backgroundTraffic;
+
+    public HapiSpecWaitUntilNextBlock withBackgroundTraffic(final boolean backgroundTraffic) {
+        this.backgroundTraffic = backgroundTraffic;
+        return this;
+    }
+
+    @Override
+    protected boolean submitOp(@NonNull final HapiSpec spec) throws Throwable {
+        final var blockDir = spec.targetNetworkOrThrow().nodes().getFirst().getExternalPath(BLOCK_STREAMS_DIR);
+        if (blockDir == null) {
+            throw new IllegalStateException("Block stream directory not available");
+        }
+
+        // Ensure the directory exists before trying to walk it
+        if (!Files.exists(blockDir)) {
+            log.info("Creating block stream directory at {}", blockDir);
+            Files.createDirectories(blockDir);
+        }
+
+        final var currentBlock = findLatestBlockNumber(blockDir);
+        final var targetBlock = currentBlock + 1;
+
+        log.info("Waiting for block {} to appear (current block is {})", targetBlock, currentBlock);
+
+        // Start background traffic if configured
+        final var stopTraffic = new AtomicBoolean(false);
```

Comment:
We don't need background traffic for blocks to advance (**unlike** with the record stream, which required user transactions to close a record file).

So we can delete this and simplify a bit.